### PR TITLE
adding sarif sdk submodule

### LINKED ISSUE #324: Enlist in SARIF-SDK as a sub-module
BinSkim is a prominent (and maybe the only, external to SARIF SDK itself!) consumer of the SARIF SDK driver framework. As such, it reveals many bugs, new feature opportunities, etc. We should enlist in the SARIF-SDK as a submodule and compile/maintain it in place with BinSkim. We've had a lot of success with this approach with other tooling, looks like it works well.

@eddynaka

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #324

### CODE REVIEW

File: src/BinSkim.Rules/BinarySkimmer.cs [hunk 1]
```diff
@@ -28,11 +28,11 @@ public BinarySkimmer()
...
                 this.SetProperty<string>(BinScopeCompatibility.EquivalentBinScopeRulePropertyName, altId);
             }
-        }
 
-        // BA2012.DoNotModifyStackProtectionCookie is the only rule
-        // that potentially fires warnings (and its default is error).
-        public override FailureLevel DefaultLevel => FailureLevel.Error;
+            // BA2012.DoNotModifyStackProtectionCookie is the only rule
+            // that potentially fires warnings (and its default is error).
+            this.DefaultConfiguration.Level = FailureLevel.Error;
```
Comment (michaelcfanning) [inline src/BinSkim.Rules/BinarySkimmer.cs:34]: >FailureLevel.Error [](start = 46, length = 18)

FYI - this code is changed in BA2004, so we need to watch the merge carefully. The default needs to be warning for now, can explain further.

Comment (eddynaka) [reply src/BinSkim.Rules/BinarySkimmer.cs:34]: I saw the change. i will do a manually merge and change to warning just like you did :)

---
In reply to: [551059054](https://github.com/microsoft/binskim/pull/325#discussion_r551059054) [](ancestors = 551059054)

**Comment by michaelcfanning** [comment]
Eddy, can we just make sure that --trace PdbLoad is still working? Looks great, this change!

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**